### PR TITLE
Moving new volumes to the encrypted directory

### PR DESCRIPTION
**Pull request by zed-rishabh**
1. After these changes, same directory will be used to store VM and container volumes i.e ```persist/vault/volumes```
2. We are passing the volume directory (to be used for storing volume) in the volume config from the ```zedagent```.

### CODE REVIEW

File: pkg/pillar/cmd/volumemgr/initialvolumestatus.go [hunk 1]
```diff
@@ -86,8 +76,9 @@ func populateInitialVolumeStatus(ctx *volumemgrContext, dirName string) {
 		volumeUUID, err := uuid.FromString(volumeID)
 		if err != nil {
 			log.Errorf("populateInitialVolumeStatus: Invalid volume UUIDStr(%s) in "+
-				"filename (%s). err: %s. XXX ignoring the File",
+				"filename (%s). err: %s. Deleting the File",
 				volumeID, filelocation, err)
+			deleteFile(filelocation)
```
Comment (eriknordmark) [inline pkg/pillar/cmd/volumemgr/initialvolumestatus.go:81]: This is still doing to mess up the old files, no? Or is the assumption that this delete only happens in /persist/vault/volumes?

File: pkg/pillar/cmd/volumemgr/volumemgr.go [hunk 2]
```diff
@@ -22,11 +22,12 @@ import (
...
-	agentName        = "volumemgr"
-	runDirname       = "/var/run/" + agentName
-	ciDirname        = runDirname + "/cloudinit" // For cloud-init volumes XXX change?
-	rwImgDirname     = types.RWImgDirname        // We store volumes here
-	roContImgDirname = types.ROContImgDirname    // We store container bootable disks here
+	agentName              = "volumemgr"
+	runDirname             = "/var/run/" + agentName
+	ciDirname              = runDirname + "/cloudinit"    // For cloud-init volumes XXX change?
+	rwImgDirname           = types.RWImgDirname           // We store volumes here
+	roContImgDirname       = types.ROContImgDirname       // We store container bootable disks here
```
Comment (eriknordmark) [inline pkg/pillar/cmd/volumemgr/volumemgr.go:29]: Do you still need these? Are their comments still correct?

File: pkg/pillar/cmd/volumemgr/volumemgr.go [hunk 3]
```diff
@@ -22,11 +22,12 @@ import (
...
+	volumeEncryptedDirName = types.VolumeEncryptedDirName // We store container bootable disks here
```
Comment (eriknordmark) [inline pkg/pillar/cmd/volumemgr/volumemgr.go:30]: Shouldn't the comment be saying "VM and OCI volumes here"?

File: pkg/pillar/cmd/volumemgr/initialvolumestatus.go (hunk 1 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/volumemgr/initialvolumestatus.go:81]: No, delete will only happen in /persist/vault/volumes

File: pkg/pillar/cmd/volumemgr/volumemgr.go (hunk 2 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/volumemgr/volumemgr.go]: We need rwImgDirname and roContImgDirname to old volumes. Corrected the comments.

File: pkg/pillar/cmd/volumemgr/volumemgr.go (hunk 3 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/volumemgr/volumemgr.go]: Corrected.

File: pkg/pillar/cmd/volumemgr/initialvolumestatus.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/volumemgr/initialvolumestatus.go:81]: OK

File: pkg/pillar/cmd/volumemgr/volumemgr.go (hunk 2 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/volumemgr/volumemgr.go]: OK

**Pull request merged**